### PR TITLE
Maven archetype git

### PR DESCRIPTION
**Pull request by bkumaran**
Creae pull request for mvn archetype/git integration

### CODE REVIEW

**Comment by bkumaran** [comment]
Adding reviewers

File: main.go [hunk 1]
```diff
@@ -188,7 +180,106 @@ func getServiceCatalogFlag() []cli.Flag {
...
+	    	cmdOut2 []byte
+	    	err2    error
+	    	cmdOut3 []byte
+	    	err3    error
+	    	cmdOut4 []byte
+	    	err4    error
+	)
+
+	// Run mvn archetype command to create a poseidon directory
+	if mvnArchetypeOut, mvnArchetypeerr = exec.Command("mvn","archetype:generate","-DarchetypeRepository=http://artifactory.nm.flipkart.com:8081/artifactory/libs-snapshots-local","-DarchetypeGroupId=com.flipkart.poseidon.apps.sample","-DarchetypeArtifactId=application-archetype", "-DarchetypeVersion=1.0.0-SNAPSHOT","-DgroupId=com.flipkart.poseidon.apps","-Dversion=1.0.0-SNAPSHOT","-DartifactId="+appName,"-Dpackage=com.flipkart."+appName,"-DApplication="+appName).Output(); mvnArchetypeerr != nil {
```
Comment (hemantj) [inline main.go]: Read defaults from the config file(yml)

Comment (hemantj) [inline main.go]: I help show user the defaults being used and informing them they can override them

File: main.go [hunk 2]
```diff
@@ -188,7 +180,106 @@ func getServiceCatalogFlag() []cli.Flag {
...
+
+  	// pwd for debugging purposes only
+	if pwdOut, pwderr = exec.Command("pwd").Output(); pwderr != nil {
+		fmt.Fprintln(os.Stderr, "There was an error runing pwd: ", pwderr)
+		os.Exit(1)
+	}
+	pwdoutput := string(pwdOut)
+	fmt.Println("pwd output : ", pwdoutput)
+
+  	// create a new git repository
```
Comment (hemantj) [inline main.go]: Pushing code to git hub should be a helper method as it can be used with other containers also.

**Comment by bkumaran** [comment]
Made the changes

File: main.go (hunk 1 above)
Comment (bkumaran) [inline main.go]: Only this i didn't find appropriate way.

**Comment by hemantj** [comment]
LGTM

**Pull request merged**